src/corehost/cli/fxr/fx_resolver.cpp

```diff
@@ -10,124 +10,170 @@ namespace
 {
     const int Max_Framework_Resolve_Retries = 100;
 
-    fx_ver_t resolve_framework_version(
+    fx_ver_t search_for_best_framework_match(
```

Comment:
The functions in this file are really large. I have been told that if a function is more than ~ten lines it needs to be refactored.  Idealistically, the code should really read like pseudo code.

I can't say I have thoroughly reviewed this file, but I can say it is less than ideal.

Reply:
OK - I'll look into it. But honestly 10 lines of C++ is basically a one liner in C#, so it's not much functionality. My personal rule of thumb is that one function should roughly fit onto a screen (so that I don't have to scroll much to read it). This is obviously subjective but it basically means something around 60-70 lines. Which in really means something like 10-20 meaningful statements which sounds reasonable to me.

Reply:
I split some of them into smaller chunks, hopefully it's easier to read now.